kong/plugins/acl/handler.lua

```diff
@@ -30,9 +30,15 @@ function ACLHandler:access(conf)
   ACLHandler.super.access(self)
 
   local consumer_id
-  if ngx.ctx.authenticated_credential then
+  if ngx.ctx.authenticated_consumer then
+    consumer_id = ngx.ctx.authenticated_consumer.id
+  end
+
+  if not consumer_id and ngx.ctx.authenticated_credential then
     consumer_id = ngx.ctx.authenticated_credential.consumer_id
-  else
+  end
+
```

Comment:
Actually no. There is always change that `ngx.ctx.authenticated_credential` is set but ` `ngx.ctx.authenticated_credential.customer_id` results `nil`. For example right now I'm working this around by setting `ngx.ctx.authenticated_credential.customer_id`, but that feels wrong as I cannot set `ngx.ctx.authenticated_credential.id` as there is no credential. My approach works even if people do stupid things, and it doesn't hurt anything (maybe just one extra `nil` check).